src/ert/storage/local_ensemble.py

```diff
@@ -557,44 +595,36 @@ def load_parameters(
         otherwise it will return the raw values.
 
         """
-        if group not in self.experiment.parameter_configuration:
+        if group in self.experiment.scalar_groups:
+            return self._load_scalar_keys(
+                self.experiment.scalar_groups[group], realizations, transformed
+            )
+        elif group not in self.experiment.parameter_configuration:
             raise KeyError(f"{group} is not registered to the experiment.")
-        config = self.experiment.parameter_configuration[group]
-        if isinstance(config, GenKwConfig):
-            df_lazy = self._load_parameters_lazy(group)
-            if realizations is not None:
-                if isinstance(realizations, int):
-                    realizations = np.array([realizations])
-                df_lazy = df_lazy.filter(pl.col("realization").is_in(realizations))
-            df = df_lazy.collect(engine="streaming")
-            if df.is_empty():
-                raise IndexError(
-                    f"No matching realizations {realizations} found for {group}"
-                )
-            if transformed:
-                df = df.with_columns(
-                    [
-                        pl.col(col)
-                        .map_elements(
-                            config.transform_col(col), return_dtype=df[col].dtype
-                        )
-                        .alias(col)
-                        for col in df.columns
-                        if col != "realization"
-                    ]
-                )
-            return df
+        if group in self.experiment.scalar_nodes:
+            return self._load_scalar_keys([group], realizations, transformed)
         ds = self._load_dataset(
             group,
-            realizations
-            if realizations is not None
-            else np.flatnonzero(self.get_realization_mask_with_parameters()),
+            (
+                realizations
+                if realizations is not None
+                else np.flatnonzero(self.get_realization_mask_with_parameters())
+            ),
         )
         return ds
 
     def load_parameters_numpy(
         self, group: str, realizations: npt.NDArray[np.int_]
     ) -> npt.NDArray[np.float64]:
+        if group in self.experiment.scalar_groups:
+            return (
+                self._load_scalar_keys(
+                    self.experiment.scalar_groups[group], realizations
+                )
+                .drop("realization")
+                .to_numpy()
+                .T.copy()
```

Comment:
Is the `copy` here necessary?

Reply:
yes, that is necessary as we do in place numpy updates.